tests/app/test_dependencies.py

```diff
@@ -361,3 +379,105 @@ async def test_get_cell_types_kg_hierarchy(
     )
 
     assert os.path.exists(settings.knowledge_graph.ct_saving_path)
+
+
+def test_get_connection_string_full(monkeypatch, patch_required_env):
+    monkeypatch.setenv("NEUROAGENT_DB__PREFIX", "http://")
+    monkeypatch.setenv("NEUROAGENT_DB__USER", "John")
+    monkeypatch.setenv("NEUROAGENT_DB__PASSWORD", "Doe")
+    monkeypatch.setenv("NEUROAGENT_DB__HOST", "localhost")
+    monkeypatch.setenv("NEUROAGENT_DB__PORT", "5000")
+    monkeypatch.setenv("NEUROAGENT_DB__NAME", "test")
+    monkeypatch.setenv("NEUROAGENT_KEYCLOAK__USERNAME", "fake_username")
+    monkeypatch.setenv("NEUROAGENT_KEYCLOAK__PASSWORD", "fake_password")
+
+    settings = Settings()
+    result = get_connection_string(settings)
+    assert (
+        result == "http://John:Doe@localhost:5000/test"
+    ), "must return fully formed connection string"
+
+
+def test_get_connection_string_no_prefix(monkeypatch, patch_required_env):
+    monkeypatch.setenv("NEUROAGENT_DB__PREFIX", "")
+    monkeypatch.setenv("NEUROAGENT_KEYCLOAK__USERNAME", "fake_username")
+    monkeypatch.setenv("NEUROAGENT_KEYCLOAK__PASSWORD", "fake_password")
+
+    settings = Settings()
+
+    result = get_connection_string(settings)
+    assert result is None, "should return None when prefix is not set"
+
+
+@patch("neuroagent.app.dependencies.create_engine")
+def test_get_engine(create_engine_mock, monkeypatch, patch_required_env):
+    create_engine_mock.return_value = Mock()
```

Comment:
It is not needed to mock the engine, there is a DB in the CI. 
To test locally. there is the command to spin up the DB in docker in `conftest.py` at line 70.